### PR TITLE
Improve architecture documentation

### PR DESCRIPTION
**Pull request by rodopoulos**
Here, we improve architecture documentation by:

- Better explaining Maestro CRDs, GRUs and schedulers
- Better improving service architecture components, focusing on explaining resources responsabilities
- Add ecosystem section, moving related content theree
- Add architecture diagram

### CODE REVIEW

File: docs/overview.md [hunk 1]
```diff
@@ -1,54 +1,120 @@
...
 
-Maestro is a Kubernetes Game Room Scheduler.
+Maestro is a Kubernetes-native game server scheduler.
 
 ## Goal
 
-Have an unified system that automatically scales game rooms regardless of the protocol (TCP, UDP). This system is related to a matchmaker but does not handle the specificities of a match such as how many players fit in a room. It only deals with high level room occupation, i.e. is the room occupied or available. The rooms communicate directly with the matchmaker in order to register and unregister themselves from the matchmaking.
+Maestro goal is to provide an unified system that automatically scales game
+rooms, regardless of the transport layer protocol (TCP, UDP). It's also
+responsible for dealing with
```
Comment (victor-carvalho) [inline docs/overview.md:10]: I think there should be something after this?

File: docs/overview.md [hunk 2]
```diff
@@ -1,54 +1,120 @@
...
-The controller is responsible for managing the Game Room Unities (GRUs). It creates and gracefully terminates GRUs according to auto scaling policies defined by the user. It makes use of the Kubernetes cluster's API endpoints in order to have updated information about the GRUs managed by Maestro. It is also responsible for persisting relevant information in the database and managing rooms statuses.
+The API is the connection of Maestro to the external world and with the GRUs
+themselves. It is responsible for:
 
-### maestro-watcher
+- Receiving and processing client operations requests over schedulers;
+- Listening to [GRU status](#room-protocol), through healthchecks;
+- Storing scheduler configurations in the database. They will be consulted by
+  the other entities, like the `watcher`;
+- Managing the pool of GRUs, keeping track of each host IP and port.
```
Comment (victor-carvalho) [inline docs/overview.md:71]: This is done inside the worker by the watcher that keeps an `watch` on Kubernetes pod resources and updates Redis with this information.

File: docs/overview.md [hunk 3]
```diff
@@ -1,54 +1,120 @@
...
 
-The watcher ensures that at any time the Game Room Unities (GRUs) state is as expected. If the scaling policies say that one should have 10 GRUs of a given type, the watcher will ask the controller to create or terminate GRUs as needed. The desired state is kept in a database that is consulted by the watcher (via controller) each time it runs. It has a lock so Maestro can be scaled horizontally. Each scheduler (i.e. maestro scalable entity) has its own watcher.
+Then, we have **maestro-worker**. They are instanced on demand,
```
Comment (victor-carvalho) [inline docs/overview.md:73]: Right now we don't have any concept of creating workers on demand. We already had some discussions about using a different model but we already have to have at least one worker running and recently we are incentivising only having one instance of the worker.

Comment (rodopoulos) [reply docs/overview.md]: Changed it. PTAL?

File: docs/overview.md (hunk 2 above)
Comment (rodopoulos) [reply docs/overview.md]: Fixed.

File: docs/overview.md (hunk 1 above)
Comment (rodopoulos) [reply docs/overview.md:10]: Fixed.

**Review by gabrielcorado — APPROVED** [review]

**Review by victor-carvalho — APPROVED** [review]

**Pull request merged**